scaleway/helpers_vpcgw.go

```diff
@@ -36,3 +37,33 @@ func vpcgwAPIWithZoneAndID(m interface{}, id string) (*vpcgw.API, scw.Zone, stri
 	}
 	return vpcgwAPI, zone, ID, nil
 }
+
+type vpcHandler struct {
+	ctx  context.Context
+	api  *vpcgw.API
+	zone scw.Zone
+}
+
+func newVPCHandler(ctx context.Context, api *vpcgw.API, zone scw.Zone) *vpcHandler {
+	return &vpcHandler{ctx: ctx, api: api, zone: zone}
+}
+
+func (vpcH *vpcHandler) waitGNetwork(gwNetworkID string) (*vpcgw.GatewayNetwork, error) {
+	retryInterval := retryIntervalVPCGatewayNetwork
```

Comment:
Have this code directly inline. We need to stay consistent with the rest of the wait operation across the provider.

Reply:
We will tackle all the issues related to waiting in a single PR in the future that will add consistent support across all resources. This PR will also handle retry and timeout.

Reply:
Updated on last commit